cpplint failing due to [build/include] warnings

`[build/include]` is disabled in `CPPLINT.cfg` pending the resolution of this issue.

```
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/MathematicalProgram.cpp:0: warning: drake/solvers/MathematicalProgram.cpp should include its header file drake/solvers/MathematicalProgram.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/MobyLCP.cpp:6: warning: drake/solvers/MobyLCP.cpp should include its header file drake/solvers/MobyLCP.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/ContinuityConstraint.cpp:2: warning: drake/solvers/qpSpline/ContinuityConstraint.cpp should include its header file drake/solvers/qpSpline/ContinuityConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/SplineInformation.cpp:2: warning: drake/solvers/qpSpline/SplineInformation.cpp should include its header file drake/solvers/qpSpline/SplineInformation.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/ValueConstraint.cpp:2: warning: drake/solvers/qpSpline/ValueConstraint.cpp should include its header file drake/solvers/qpSpline/ValueConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/controllers/InstantaneousQPController.cpp:2: warning: drake/systems/controllers/InstantaneousQPController.cpp should include its header file drake/systems/controllers/InstantaneousQPController.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/collision/Element.cpp:3: warning: drake/systems/plants/collision/Element.cpp should include its header file drake/systems/plants/collision/Element.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/collision/Model.cpp:3: warning: drake/systems/plants/collision/Model.cpp should include its header file drake/systems/plants/collision/Model.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/constraint/constructPtrRigidBodyConstraint.cpp:1: warning: drake/systems/plants/constraint/constructPtrRigidBodyConstraint.cpp should include its header file drake/systems/plants/constraint/constructPtrRigidBodyConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/HelicalJoint.cpp:0: warning: drake/systems/plants/joints/HelicalJoint.cpp should include its header file drake/systems/plants/joints/HelicalJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/PrismaticJoint.cpp:0: warning: drake/systems/plants/joints/PrismaticJoint.cpp should include its header file drake/systems/plants/joints/PrismaticJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/QuaternionFloatingJoint.cpp:2: warning: drake/systems/plants/joints/QuaternionFloatingJoint.cpp should include its header file drake/systems/plants/joints/QuaternionFloatingJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/RevoluteJoint.cpp:0: warning: drake/systems/plants/joints/RevoluteJoint.cpp should include its header file drake/systems/plants/joints/RevoluteJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/RollPitchYawFloatingJoint.cpp:2: warning: drake/systems/plants/joints/RollPitchYawFloatingJoint.cpp should include its header file drake/systems/plants/joints/RollPitchYawFloatingJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/RigidBody.cpp:4: warning: drake/systems/plants/RigidBody.cpp should include its header file drake/systems/plants/RigidBody.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/rigidBodyTreeMexFunctions.cpp:1: warning: drake/systems/plants/rigidBodyTreeMexFunctions.cpp should include its header file drake/systems/plants/rigidBodyTreeMexFunctions.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/Element.cpp:0: warning: drake/systems/plants/shapes/Element.cpp should include its header file drake/systems/plants/shapes/Element.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/Geometry.cpp:1: warning: drake/systems/plants/shapes/Geometry.cpp should include its header file drake/systems/plants/shapes/Geometry.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/VisualElement.cpp:0: warning: drake/systems/plants/shapes/VisualElement.cpp should include its header file drake/systems/plants/shapes/VisualElement.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/xmlUtil.cpp:2: warning: drake/systems/plants/xmlUtil.cpp should include its header file drake/systems/plants/xmlUtil.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/robotInterfaces/BodyMotionData.cpp:0: warning: drake/systems/robotInterfaces/BodyMotionData.cpp should include its header file drake/systems/robotInterfaces/BodyMotionData.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/trajectories/PiecewiseFunction.cpp:2: warning: drake/systems/trajectories/PiecewiseFunction.cpp should include its header file drake/systems/trajectories/PiecewiseFunction.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/util/yaml/yamlUtil.cpp:0: warning: drake/util/yaml/yamlUtil.cpp should include its header file drake/util/yaml/yamlUtil.h  [build/include] [5]
```

--- a/CPPLINT.cfg
+++ b/CPPLINT.cfg
@@ -15,9 +15,6 @@ filter=-build/c++11
 # Disable header_guard warning (see Drake issue #2104).
 filter=-build/header_guard
 
-# TODO(#2364) Fix this.
-filter=-build/include
-
 # TODO(#2269) Fix this.
 filter=-build/include_order
 

--- a/drake/automotive/gen/driving_command_translator.h
+++ b/drake/automotive/gen/driving_command_translator.h
@@ -3,6 +3,8 @@
 // GENERATED FILE DO NOT EDIT
 // See drake/tools/lcm_vector_gen.py.
 
+#include <vector>
+
 #include "drake/automotive/gen/driving_command.h"
 #include "drake/common/drake_export.h"
 #include "drake/lcmt_driving_command_t.hpp"

--- a/drake/automotive/gen/euler_floating_joint_state_translator.h
+++ b/drake/automotive/gen/euler_floating_joint_state_translator.h
@@ -3,6 +3,8 @@
 // GENERATED FILE DO NOT EDIT
 // See drake/tools/lcm_vector_gen.py.
 
+#include <vector>
+
 #include "drake/automotive/gen/euler_floating_joint_state.h"
 #include "drake/common/drake_export.h"
 #include "drake/lcmt_euler_floating_joint_state_t.hpp"

--- a/drake/automotive/gen/idm_with_trajectory_agent_state_translator.h
+++ b/drake/automotive/gen/idm_with_trajectory_agent_state_translator.h
@@ -3,6 +3,8 @@
 // GENERATED FILE DO NOT EDIT
 // See drake/tools/lcm_vector_gen.py.
 
+#include <vector>
+
 #include "drake/automotive/gen/idm_with_trajectory_agent_state.h"
 #include "drake/common/drake_export.h"
 #include "drake/lcmt_idm_with_trajectory_agent_state_t.hpp"

--- a/drake/automotive/gen/simple_car_config_translator.h
+++ b/drake/automotive/gen/simple_car_config_translator.h
@@ -3,6 +3,8 @@
 // GENERATED FILE DO NOT EDIT
 // See drake/tools/lcm_vector_gen.py.
 
+#include <vector>
+
 #include "drake/automotive/gen/simple_car_config.h"
 #include "drake/common/drake_export.h"
 #include "drake/lcmt_simple_car_config_t.hpp"

--- a/drake/automotive/gen/simple_car_state_translator.h
+++ b/drake/automotive/gen/simple_car_state_translator.h
@@ -3,6 +3,8 @@
 // GENERATED FILE DO NOT EDIT
 // See drake/tools/lcm_vector_gen.py.
 
+#include <vector>
+
 #include "drake/automotive/gen/simple_car_state.h"
 #include "drake/common/drake_export.h"
 #include "drake/lcmt_simple_car_state_t.hpp"

--- a/drake/automotive/idm_with_trajectory_agent.cc
+++ b/drake/automotive/idm_with_trajectory_agent.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/automotive/idm_with_trajectory_agent-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/automotive/simple_car.cc
+++ b/drake/automotive/simple_car.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/automotive/simple_car-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/examples/bouncing_ball/ball.cc
+++ b/drake/examples/bouncing_ball/ball.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/examples/bouncing_ball/ball-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/examples/bouncing_ball/bouncing_ball.cc
+++ b/drake/examples/bouncing_ball/bouncing_ball.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/examples/bouncing_ball/bouncing_ball-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/matlab/CPPLINT.cfg
+++ b/drake/matlab/CPPLINT.cfg
@@ -1,4 +1,5 @@
 # These are warnings we don't care about in legacy matlab code.
+filter=-build/include
 filter=-build/namespaces
 filter=-readability/casting
 filter=-readability/namespace

--- a/drake/solvers/gurobi_solver.cc
+++ b/drake/solvers/gurobi_solver.cc
@@ -8,6 +8,7 @@
 #include <Eigen/Core>
 #include <Eigen/Sparse>
 
+// NOLINTNEXTLINE(build/include) False positive due to weird include style.
 #include "gurobi_c++.h"
 
 #include "drake/common/drake_assert.h"

--- a/drake/system1/CPPLINT.cfg
+++ b/drake/system1/CPPLINT.cfg
@@ -1,4 +1,5 @@
 # These are warnings we don't care about in legacy code.
+filter=-build/include
 filter=-build/namespaces
 filter=-readability/namespace
 filter=-runtime/references

--- a/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp
+++ b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp
@@ -1,5 +1,7 @@
 #include <lcm/lcm.h>
 #include <bot_lcmgl_client/lcmgl.h>
+
+// NOLINTNEXTLINE(build/include) Generated C code (N.B. not C++) is odd.
 #include "drake_lcmt_zmp_com_observer_state.h"
 
 bot_lcmgl_t *lcmgl;

--- a/drake/systems/framework/primitives/constant_value_source.cc
+++ b/drake/systems/framework/primitives/constant_value_source.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/systems/framework/primitives/constant_value_source-inl.h"
 
 namespace drake {

--- a/drake/systems/framework/primitives/gain.cc
+++ b/drake/systems/framework/primitives/gain.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/systems/framework/primitives/gain-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/systems/framework/primitives/pass_through.cc
+++ b/drake/systems/framework/primitives/pass_through.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/systems/framework/primitives/pass_through-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/systems/framework/primitives/zero_order_hold.cc
+++ b/drake/systems/framework/primitives/zero_order_hold.cc
@@ -1,3 +1,4 @@
+// NOLINTNEXTLINE(build/include) False positive on inl file.
 #include "drake/systems/framework/primitives/zero_order_hold-inl.h"
 
 #include "drake/common/eigen_autodiff_types.h"

--- a/drake/tools/lcm_vector_gen.py
+++ b/drake/tools/lcm_vector_gen.py
@@ -157,6 +157,8 @@ TRANSLATOR_HH_PREAMBLE = """
 
 %(generated_code_warning)s
 
+#include <vector>
+
 #include "%(relative_cxx_dir)s/%(snake)s.h"
 #include "drake/common/drake_export.h"
 #include "drake/systems/lcm/lcm_and_vector_base_translator.h"
@@ -302,7 +304,7 @@ def generate_code(args):
 
     namespace = args.namespace.split("::")
     opening_namespace = "".join(["namespace " + x + "{\n" for x in namespace])
-    closing_namespace = "".join(["}  // namespace " + x + "\n" 
+    closing_namespace = "".join(["}  // namespace " + x + "\n"
                                  for x in reversed(namespace)])
 
     if args.yaml_file:

--- a/drake/util/CPPLINT.cfg
+++ b/drake/util/CPPLINT.cfg
@@ -1,4 +1,5 @@
 # These are warnings we don't care about in legacy matlab code.
+filter=-build/include
 filter=-build/namespaces
 filter=-readability/namespace
 filter=-runtime/references

